tests/unit/jobserver/test_icons.py

```diff
@@ -1,30 +1,38 @@
-import itertools
 import re
 from pathlib import Path
 
+import pytest
 
-def test_all_icons(subtests):
-    """Ensure icon tag names defined in templates are valid tags"""
-    templates = [p for p in Path("templates").glob("**/*.html")]
 
-    # remove templates with no icon tag so we can simplify the regex matching
-    # code below
-    icon_templates = [p for p in templates if "{% icon_" in p.read_text()]
-
-    # look for the start of an icon templatetag and capture the full name
-    icon_pat = re.compile(r"{% (icon_[^\s]+)")
-
-    # get a set of icon names in the Django templates
-    icons = set(
-        itertools.chain.from_iterable(
-            icon_pat.search(t.read_text()).groups() for t in icon_templates
+# Logically, this feels like it could be a pytest.fixture.
+# However, it's not currently possible to use fixtures with parametrize:
+# https://github.com/pytest-dev/pytest/issues/349
+def find_icon_tag_names_used():
+    """Find the name of template tags starting with `icon_` used in our templates."""
+    icon_pattern = re.compile(r"{% (icon_[^\s]+)")
+    template_paths = list(Path("templates").rglob("*.html"))
+    # Get a sorted list of non-duplicated icon names in the Django templates.
+    # We use a sorted list so that the different xdist workers get a consistent ordering.
+    # https://pytest-xdist.readthedocs.io/en/stable/known-limitations.html#order-and-amount-of-test-must-be-consistent
+    return sorted(
+        set(
+            icon_tag_name
+            for path in template_paths
+            for icon_tag_name in icon_pattern.findall(path.read_text())
         )
     )
 
-    # read the component config contents into memory so we don't have to open
-    # it for each icon
-    config = Path("templates/components.yaml").read_text()
-    for icon in icons:
-        # use subtests to avoid failing the test at the first broken icon
-        with subtests.test(icon=icon):
-            assert icon in config
+
+@pytest.fixture(scope="module")
+def component_config():
+    # Note: this fixture does not get reused with pytest-xdist,
+    # so the file is read once per-worker, but this is very fast anyway.
+    # https://github.com/pytest-dev/pytest-xdist/issues/783
+    return Path("templates/components.yaml").read_text()
+
+
+@pytest.mark.parametrize("icon_tag_name", find_icon_tag_names_used())
+def test_icon_tag_names_config(icon_tag_name, component_config):
+    """Ensure icon tag names defined in templates appear in the icon component
+    configuration file"""
```

Comment:
```suggestion
    configuration file."""
```
For punctuation consistency.

Reply:
Done.